crates/vm/src/system/native_adapter/mod.rs

```diff
@@ -146,6 +162,20 @@ impl<AB: InteractionBuilder, const R: usize, const W: usize> VmAdapterAir<AB>
     }
 }
 
+#[repr(C)]
+#[derive(AlignedBytesBorrow, Debug)]
+pub struct NativeAdapterRecord<F, const R: usize, const W: usize> {
```

Comment:
for Native, the more optimal thing I had in mind was not to use `Record` but only have `RecordMut` which would give pointers to the locations of field elements in-place. 
However since this is already implemented, and it's simpler this way for the dense record implementation, we can save it for later